1. Faça um Programa que peça dois números e imprima o maior deles.

In [1]:
n1 = int(input("Digite um numero: "))
n2 = int(input("Digite outro numero: "))

if n1 > n2:
    print(f"Maior numero {n1}")
else:
    print(f"Maior numero {n2}")

Digite um numero: 2
Digite outro numero: 5
Maior numero 5


2. Faça um Programa que verifique se uma letra digitada é vogal ou consoante.

In [3]:
vogal = ['A', 'E', 'I', 'O', 'U']

letra = input("Digite uma letra: ")

if letra.upper() in vogal:
    print(f"A letra {letra} e uma vogal")
else:
    print(f"A letra {letra} nao e uma vogal")

Digite uma letrap


3. Faça um programa para a leitura de duas notas parciais de um aluno. O programa
deve calcular a média alcançada por aluno e apresentar:
- A mensagem "Aprovado", se a média alcançada for maior ou igual a sete;
- A mensagem "Reprovado", se a média for menor do que sete;
- A mensagem "Aprovado com Distinção", se a média for igual a dez.

In [10]:
nota1 = 5
nota2 = 10

media = (nota1 + nota2 )/ 2

if media >= 7 and media < 10:
    print("Aprovado")
elif media <= 5:
    print("Reprovado")
else:
    print("Aprovado com Distinção")

7.5
Aprovado


4. Faça um Programa que leia três números e mostre-os em ordem decrescente.

In [19]:
lista_numeros = list()

n1 = 5
n2 = 2
n3 = 7

lista_numeros = [n1, n2, n3]
lista_numeros = sorted(lista_numeros)[::-1]
lista_numeros = list(map(str, lista_numeros))

print(' '.join(lista_numeros))


7 5 2


5. As Organizações Tabajara resolveram dar um aumento de salário aos seus
colaboradores e lhe contrataram para desenvolver o programa que calculará os
reajustes.
Faça um programa que recebe o salário de um colaborador e o reajuste segundo o
seguinte critério, baseado no salário atual:
- salários até R$ 280,00 (incluindo) : aumento de 20%
- salários entre R$ 280,00 e R$ 700,00 : aumento de 15%
- salários entre R$ 700,00 e R$ 1500,00 : aumento de 10%
- salários de R$ 1500,00 em diante : aumento de 5% Após o aumento ser realizado,
informe na tela:
- o salário antes do reajuste;
- o percentual de aumento aplicado;
- o valor do aumento;
- o novo salário, após o aumento.

In [22]:
salario = 1600
aumento = 0
salario_total = 0
percentual = ""

if salario <= 280:
    aumento = salario*0.20
    salario_total = salario + aumento
    percentual = "20%"
elif salario > 280 and salario <= 700:
    aumento = salario*0.20
    salario_total = salario + aumento
    percentual = "20%"
elif salario > 700 and salario <= 1500:
    aumento = salario*0.10
    salario_total = salario + aumento
    percentual = "10%"
elif salario > 1500:
    aumento = salario*0.05
    salario_total = salario + aumento
    percentual = "5%"

print(f"{salario} {aumento} {salario_total} {percentual}")


1600 80.0 1680.0 5%


6. Faça um Programa que leia um número e exiba o dia correspondente da semana.
(1-Domingo, 2- Segunda, etc.), se digitar outro valor deve aparecer valor inválido.

In [24]:
dias_semana = {1:'Domingo', 2:'Segunda', 3:'Terca', 4:'Quarta', 5:'Quinta', 6:'Sexta', 7:'Sabado'}

dia = int(input("Digite um numero correspondente ao dia da semana: "))

if dia in dias_semana.keys():
    print(dias_semana[dia])
else:
    print("Invalido")

Digite um numero correspondente ao dia da semana: 8
Invalido


7. Faça um programa que lê as duas notas parciais obtidas por um aluno numa
disciplina ao longo de um semestre, e calcule a sua média. A atribuição de conceitos
obedece à tabela abaixo:
Média de Aproveitamento Conceito
Entre 9.0 e 10.0 A
Entre 7.5 e 9.0 B
Entre 6.0 e 7.5 C
Entre 4.0 e 6.0 D
Entre 4.0 e zero E
O algoritmo deve mostrar na tela as notas, a média, o conceito correspondente e a
mensagem “APROVADO” se o conceito for A, B ou C ou “REPROVADO” se o conceito
for D ou E.

In [29]:
n1 = int(input("Digite o valor da primeira nota: "))
n2 = int(input("Digite o valor da segunda nota: "))

nota = (n1 + n2)/2
print(nota)

if nota >= 9 and nota <= 10:
    print( "A nota da primeira prova: ",n1)
    print( "A nota da segunda prova: ",n2)
    print( "——————————")
    print( "Você tirou um A!")
    print( "Sua media é de:",nota)
    print( "você foi aprovado!!")

elif nota >= 7.5 and nota < 9:
    print( "A nota da primeira prova: ",n1)
    print( "A nota da segunda prova: ",n2)
    print( "——————————")
    print( "Você tirou um B!")
    print( "Sua media é de:",nota)
    print( "você foi aprovado!!")

elif nota >= 6.5 and nota < 7.5:
    print( "A nota da primeira prova: ",n1)
    print( "A nota da segunda prova: ",n2)
    print( "——————————")
    print( "Você tirou um C!")
    print( "Sua media é de:",nota)
    print( "você foi aprovado!!")

elif nota >= 4 and nota < 6:
    print( "A nota da primeira prova: ",n1)
    print( "A nota da segunda prova: ",n2)
    print( "——————————")
    print( "Você tirou um D!")
    print( "Sua media é de:",nota)
    print( "você foi reprovado!!")

elif nota < 4:
    print( "A nota da primeira prova: ",n1)
    print( "A nota da segunda prova: ",n2)
    print( "——————————")
    print( "Você tirou um E!")
    print( "Sua media é de:",nota)
    print( "você foi reprovado!!")

Digite o valor da primeira nota: 5
Digite o valor da segunda nota: 10
7.5
A nota da primeira prova:  5
A nota da segunda prova:  10
——————————
Você tirou um B!
Sua media é de: 7.5
você foi aprovado!!


8. Faça um Programa que peça os 3 lados de um triângulo. O programa deverá
informar se os valores podem ser um triângulo. Indique, caso os lados formem um
triângulo, se o mesmo é: equilátero, isósceles ou escaleno.
Dicas:
- Três lados formam um triângulo quando a soma de quaisquer dois lados for maior
que o terceiro;
- Triângulo Equilátero: três lados iguais;
- Triângulo Isósceles: quaisquer dois lados iguais;
- Triângulo Escaleno: três lados diferentes;

In [34]:
l1 = input('Digite o tamanho do primeiro lado: ')
l2 = input('Digite o segundo lado: ')
l3 = input('Digite o segundo lado: ')

if l1 + l2 > l3:
    if l1 == l2 and l1 == l3:
        print ('Triângulo Equilátero')
    elif l1 == l2 or l2 == l3 or l1 == l3:
        print ('Triângulo Isósceles')
    elif l1 != l2 and l3 or l2 != l1 and l3 or l1 != l3:
        print ('Triângulo Escaleno')

else:
    print ('É impossivel ser um triângulo')

Digite o tamanho do primeiro lado: 5
Digite o segundo lado: 5
Digite o segundo lado: 5
Triângulo Equilátero


9. Faça um programa que calcule as raízes de uma equação do segundo grau, na
forma ax2 + bx + c. O programa deverá pedir os valores de a, b e c e fazer as
consistências, informando ao usuário nas seguintes situações:
- Se o usuário informar o valor de A igual a zero, a equação não é do segundo grau e
o programa não deve fazer pedir os demais valores, sendo encerrado;

- Se o delta calculado for negativo, a equação não possui raizes reais. Informe ao
usuário e encerre o programa;
- Se o delta calculado for igual a zero a equação possui apenas uma raiz real;
informe-a ao usuário;
- Se o delta for positivo, a equação possui duas raiz reais; informe-as ao usuário;

In [8]:
a = float(input("digite o valor de a"))
delta = True

if a == 0:
    print("Nao e 2 grau")
    delta = False
else:
    b = float(input("digite o valor de b"))
    c = float(input("digite o valor de c"))

if (delta):
    eqb = 0
    eqA = 0
    delta = (b**2)-(4*a*c)
    if delta <0:
        print("Nao possui raizes reais")
    elif delta == 0:
        print("Possui apenas uma raiz real")
    else:
        #Calcular com delta
        eqA = -b + (delta ** 0.5) / (2*a)
        eqB = -b - (delta ** 0.5) / (2*a)
        print(f"A: {eqA}, B: {eqB}")
        
 

digite o valor de a1
digite o valor de b10
digite o valor de c1
A: -5.101020514433644, B: -14.898979485566356


10. Faça um Programa para um caixa eletrônico. O programa deverá perguntar ao
usuário a valor do saque e depois informar quantas notas de cada valor serão
fornecidas. As notas disponíveis serão as de 1, 5, 10, 50 e 100 reais. O valor mínimo é
de 10 reais e o máximo de 600 reais. O programa não deve se preocupar com a
quantidade de notas existentes na máquina.
- Exemplo 1: Para sacar a quantia de 256 reais, o programa fornece duas notas de
100, uma nota de 50, uma nota de 5 e uma nota de 1;
- Exemplo 2: Para sacar a quantia de 399 reais, o programa fornece três notas de
100, uma nota de 50, quatro notas de 10, uma nota de 5 e quatro notas de 1.

In [20]:
valor = float(input("Digite o valor que voce quer sacar: "))

notas_disp = [100, 50, 10, 5, 1]
nota = []

if valor >= 10 and valor <=600:
    count = 0
    while valor > 0:
        if (valor - notas_disp[count]) >= 0:
            nota.append(notas_disp[count])
            valor = valor - notas_disp[count]
        else:
            count +=1
print(nota)
    
    

Digite o valor que voce quer sacar: 75
[50, 10, 10, 5]


11. Faça um programa que faça 5 perguntas para uma pessoa sobre um crime. As
perguntas são:
"Telefonou para a vítima?"
"Esteve no local do crime?"
"Mora perto da vítima?"
"Devia para a vítima?"
"Já trabalhou com a vítima?"
O programa deve no final emitir uma classificação sobre a participação da pessoa no
crime. Se a pessoa responder positivamente a 2 questões ela deve ser classificada
como "Suspeita", entre 3 e 4 como "Cúmplice" e 5 como "Assassino". Caso contrário,
ele será classificado como "Inocente".

In [20]:
perguntas = {'Telefonou para a vítima?':False, 'Esteve no local do crime?':False, 
             'Mora perto da vítima?':False, 'Devia para a vítima?':False, 'Já trabalhou com a vítima?':False}
for pergunta in perguntas:
    resposta = input(f"{pergunta} S/N: ").upper()
    if resposta == 'S':
        perguntas[pergunta] = True
respostas = sum(perguntas.values())

if respostas == 2:
    print("Suspeita")
elif respostas >2 and respostas <=4:
    print("Cumplice")
elif respostas >4:
    print("Assasino")
else:
    print("Inocente")
    

Telefonou para a vítima? S/NN
Esteve no local do crime? S/NN
Mora perto da vítima? S/NS
Devia para a vítima? S/NN
Já trabalhou com a vítima? S/NS
Suspeita


12. Uma fruteira está vendendo frutas com a seguinte tabela de preços:
Até 5 Kg Acima de 5 Kg
Morango R$ 2,50 por Kg R$ 2,20 por Kg
Maçã R$ 1,80 por Kg R$ 1,50 por Kg
Se o cliente comprar mais de 8 Kg em frutas ou o valor total da compra ultrapassar
R$ 25,00, receberá ainda um desconto de 10% sobre este total. Escreva um
algoritmo para ler a quantidade (em Kg) de morangos e a quantidade (em Kg) de
maças adquiridas e escreva o valor a ser pago pelo cliente.

In [30]:
frutas = {'Morango':[2.50, 2.20], 'Maca':[1.80, 1.50]}

kg_morango = 5
kg_maca = 10

kilos = [kg_morango, kg_maca]
valor_total = 0

for fruta, kg in zip(frutas, kilos):
    if kg_morango > 5 or kg_maca > 5:
        valor_total += frutas[fruta][1] * kg
    else:
        valor_total += frutas[fruta][0] * kg

if valor_total > 25:
    valor_total = valor_total - (valor_total*0.1)
    
print(f"Valor total: {valor_total}")

Valor total: 23.4


13. Faça um programa que peça uma nota, entre zero e dez. Mostre uma mensagem
caso o valor seja inválido e continue pedindo até que o usuário informe um valor
válido.

In [51]:
while True:
    nota = int(input("Digite uma nota: "))
    if (nota < 0 or nota > 10):
        nota-=1
        print("Nota invalida")
    else:
        break

Digite uma nota: 12
Nota invalida
Digite uma nota: 12
Nota invalida
Digite uma nota: 12
Nota invalida
Digite uma nota: 12
Nota invalida
Digite uma nota: 12
Nota invalida
Digite uma nota: 10


13. Faça um programa que leia e valide as seguintes informações:
Nome: maior que 3 caracteres;
Idade: entre 0 e 150;

Salário: maior que zero;
Sexo: 'f' ou 'm';
Estado Civil: 's', 'c', 'v', 'd';

In [6]:
validar = {"Nome":False, "Idade":False, "Salario":False, "Sexo":False, "Estado_Civil":False}

for k, v in validar.items():
    to_validate = input(f"Digite seu(a) {k}: ")
    if "Nome" in k:
        if len(to_validate) > 3:
            validar[k] = True
    elif "Idade" in k:
        if int(to_validate) > 0 and int(to_validate) <= 150:
            validar[k] = True
    elif "Salario" in k:
        if int(to_validate) > 0:
            validar[k] = True
    elif "Sexo" in k:
        if to_validate == "m" or to_validate == "f":
            validar[k] = True
    elif "Estado_Civil" in k:
        if to_validate in ["s", "c", "v", "d"]:
            validar[k] = True
print(validar)      

Digite seu(a) Nome: Lucas
Digite seu(a) Idade: 21
Digite seu(a) Salario: 200
Digite seu(a) Sexo: m
Digite seu(a) Estado_Civil: c


14. Faça um programa que leia 5 números e informe o maior número.

In [11]:
lista_num = list()

for i in range(0,5):
    numero = int(input("Digite um numero: "))
    lista_num.append(numero)

print(max(lista_num))

Digite um numero: 1
Digite um numero: 2
Digite um numero: 3
Digite um numero: 66
Digite um numero: 5
66


16. Faça um programa que imprima na tela apenas os números ímpares entre 1 e 50.

In [18]:
for i in range(1, 51):
    if (i%2) != 0:
        print(i)

1
3
5
7
9
11
13
15
17
19
21
23
25
27
29
31
33
35
37
39
41
43
45
47
49


17. Desenvolva um gerador de tabuada, capaz de gerar a tabuada de qualquer
número inteiro entre 1 a 10. O usuário deve informar de qual numero ele deseja ver
a tabuada. A saída deve ser conforme o exemplo abaixo:
Tabuada de 5:
5 X 1 = 5
5 X 2 = 10
...
5 X 10 = 50

In [25]:
numero = int(input("Digite o numero que voce que ver a tabuada: "))
for i in range(1, 11):
    print(f"{i} X {numero} = {i*numero}")

Digite o numero que voce que ver a tabuada: 8
1 X 8 = 8
2 X 8 = 16
3 X 8 = 24
4 X 8 = 32
5 X 8 = 40
6 X 8 = 48
7 X 8 = 56
8 X 8 = 64
9 X 8 = 72
10 X 8 = 80


18. A série de Fibonacci é formada pela seqüência 1,1,2,3,5,8,13,21,34,55,... Faça um
programa capaz de gerar a série até o n−ésimo termo.

In [40]:
n = int(input("Digite o numero: "))
fat = 1
x = 1

while x <= n:
    fat *= x
    x += 1
    
print(fat)

Digite o numero: 5
120


19. Faça um programa que calcule o fatorial de um número inteiro fornecido pelo
usuário. Ex.: 5!=5.4.3.2.1=120

In [43]:
def fibonacci(n): 
    a = 0
    b = 1
    if n < 0: 
        print("Incorrect input") 
    elif n == 0: 
        return a 
    elif n == 1: 
        return b 
    else: 
        for i in range(2,n): 
            c = a + b 
            a = b 
            b = c 
        return b 
n = int(input("Digite o numero para ver o fibonacci: "))
print(fibonacci(n)) 

Digite o numero para ver o fibonacci: 5
3


20. O Sr. Manoel Joaquim possui uma grande loja de artigos de R$ 1,99, com cerca
de 10 caixas. Para agilizar o cálculo de quanto cada cliente deve pagar ele
desenvolveu um tabela que contém o número de itens que o cliente comprou e ao
lado o valor da conta. Desta forma a atendente do caixa precisa apenas contar
quantos itens o cliente está levando e olhar na tabela de preços. Você foi contratado
para desenvolver o programa que monta esta tabela de preços, que conterá os
preços de 1 até 50 produtos, conforme o exemplo abaixo:
Lojas Quase Dois - Tabela de preços
1 - R$ 1.99
2 - R$ 3.98
...
50 - R$ 99.50

In [56]:
def montar_tabela(preco_base):
    tabela_precos = dict()
    valor = preco_base
    result = 0

    for i in range(1, 51):
        tabela_precos[i] = round((valor + result),2)
        result += valor
    return(tabela_precos)

print(montar_tabela(1.99))
print(montar_tabela(0.18))

{1: 1.99, 2: 3.98, 3: 5.97, 4: 7.96, 5: 9.95, 6: 11.94, 7: 13.93, 8: 15.92, 9: 17.91, 10: 19.9, 11: 21.89, 12: 23.88, 13: 25.87, 14: 27.86, 15: 29.85, 16: 31.84, 17: 33.83, 18: 35.82, 19: 37.81, 20: 39.8, 21: 41.79, 22: 43.78, 23: 45.77, 24: 47.76, 25: 49.75, 26: 51.74, 27: 53.73, 28: 55.72, 29: 57.71, 30: 59.7, 31: 61.69, 32: 63.68, 33: 65.67, 34: 67.66, 35: 69.65, 36: 71.64, 37: 73.63, 38: 75.62, 39: 77.61, 40: 79.6, 41: 81.59, 42: 83.58, 43: 85.57, 44: 87.56, 45: 89.55, 46: 91.54, 47: 93.53, 48: 95.52, 49: 97.51, 50: 99.5}
{1: 0.18, 2: 0.36, 3: 0.54, 4: 0.72, 5: 0.9, 6: 1.08, 7: 1.26, 8: 1.44, 9: 1.62, 10: 1.8, 11: 1.98, 12: 2.16, 13: 2.34, 14: 2.52, 15: 2.7, 16: 2.88, 17: 3.06, 18: 3.24, 19: 3.42, 20: 3.6, 21: 3.78, 22: 3.96, 23: 4.14, 24: 4.32, 25: 4.5, 26: 4.68, 27: 4.86, 28: 5.04, 29: 5.22, 30: 5.4, 31: 5.58, 32: 5.76, 33: 5.94, 34: 6.12, 35: 6.3, 36: 6.48, 37: 6.66, 38: 6.84, 39: 7.02, 40: 7.2, 41: 7.38, 42: 7.56, 43: 7.74, 44: 7.92, 45: 8.1, 46: 8.28, 47: 8.46, 48: 8.64, 49: 8

21. O Sr. Manoel Joaquim acaba de adquirir uma panificadora e pretende implantar
a metodologia da tabelinha, que já é um sucesso na sua loja de 1,99. Você foi
contratado para desenvolver o programa que monta a tabela de preços de pães, de
1 até 50 pães, a partir do preço do pão informado pelo usuário, conforme o exemplo
abaixo:
Preço do pão: R$ 0.18
Panificadora Pão de Ontem - Tabela de preços
1 - R$ 0.18
2 - R$ 0.36

...
50 - R$ 9.00

In [59]:
valor = ""
produtos = dict()
index = 1

while valor != 0.0:
    valor = float(input("Digite o valor de um produto: "))
    produtos[index] = valor
    index +=1

print(f"Total: {sum(produtos.values())}")
print(f"Produtos: {produtos}")

pagamento = float(input("Digite o valor recebido: "))
if pagamento - sum(produtos.values()) > 0:
    print(f"Troco: {pagamento - sum(produtos.values())}")
else:
    print("Chama a policia")



Digite o valor de um produto: 8.8
Digite o valor de um produto: 9
Digite o valor de um produto: 10
Digite o valor de um produto: 88
Digite o valor de um produto: 0
Total: 115.8
Digite o valor recebido: 200
Troco: 84.2


23. Foi feita uma estatística em cinco cidades brasileiras para coletar dados sobre
acidentes de trânsito. Foram obtidos os seguintes dados:
- Código da cidade;
- Número de veículos de passeio (em 1999);
- Número de acidentes de trânsito com vítimas (em 1999). Deseja-se saber:
- Qual o maior e menor índice de acidentes de transito e a que cidade pertence;
- Qual a média de veículos nas cinco cidades juntas;
- Qual a média de acidentes de trânsito nas cidades com menos de 2.000 veículos de
passeio.

In [24]:
ibge = {
        1: {"Nome":"Bauru",
           "Acidentes":5,
           "Veiculos":2000
           },
        2: {"Nome":"Minas",
           "Acidentes":500,
           "Veiculos":2000
           }
       }

acidentes_dict = dict()

menor = ibge.get(1).get("Acidentes")
maior = ibge.get(1).get("Acidentes")

veiculos = 0
acidentes_menor = 0
count_acidentes_menor = 0

for k, v in ibge.items():
    veiculos += ibge[k].get("Veiculos")
    for item, desc in v.items():
        if item == "Acidentes":
            if desc <= menor:
                acidentes_dict["Menor"] = { "NOME":ibge[k].get("Nome"), "Acidentes" : desc }
            if desc >= maior:
                acidentes_dict["Maior"] = { "NOME":ibge[k].get("Nome"), "Acidentes" : desc }
        if item == "Veiculos" and desc <= 2000:
            acidentes_menor += ibge[k].get("Acidentes")
            count_acidentes_menor +=1
            

print(acidentes_dict)
print(veiculos/len(ibge.keys()))
print(acidentes_menor/count_acidentes_menor)

        

{'Menor': {'NOME': 'Bauru', 'Acidentes': 5}, 'Maior': {'NOME': 'Minas', 'Acidentes': 500}}
2000.0
252.5


24. Faça um programa que receba o valor de uma dívida e mostre uma tabela com
os seguintes dados: valor da dívida, valor dos juros, quantidade de parcelas e valor
da parcela.
Os juros e a quantidade de parcelas seguem a tabela abaixo:
Quantidade de Parcelas % de Juros sobre o valor inicial da dívida
1 0
3 10
6 15
9 20
12 25
Exemplo de saída do programa:
Valor da Dívida Valor dos Juros Quantidade de Parcelas Valor da Parcela
R$ 1.000,00 0 1 R$ 1.000,00
R$ 1.100,00 100 3 R$ 366,00
R$ 1.150,00 150 6 R$ 191,67

In [35]:
parcelas_juros = {1:0.0, 3:0.1, 6:0.15, 9:0.2, 12:0.25}

preco_produto = float(input("Digite o preco do produto: "))
parcelas = int(input("Digite o numero de parcelas 1/3/6/9/12: "))

juros = preco_produto * parcelas_juros[parcelas]
preco_final = preco_produto + juros

valor_parcela = preco_final / parcelas

print(f"Divida: {preco_final}, Juros: {juros}, Parcelas: {parcelas}, Valor Parcela: {round(valor_parcela,2)}")

Digite o preco do produto: 100
Digite o numero de parcelas 1/3/6/9/12: 3
Divida: 110.0, Juros: 10.0, Parcelas: 3, Valor Parcela: 36.67
